File: app/src/main/java/br/com/pedrosilva/tecnonutri/presentation/presenters/FeedPresenter.java
package br.com.pedrosilva.tecnonutri.presentation.presenters;

import java.util.List;

import br.com.pedrosilva.tecnonutri.domain.entities.FeedItem;
import br.com.pedrosilva.tecnonutri.presentation.presenters.base.BasePresenter;
import br.com.pedrosilva.tecnonutri.presentation.ui.BaseView;


public interface FeedPresenter extends BasePresenter {

    interface View extends BaseView {
        void onLoadFeed(List<FeedItem> feedItems, int page, int timestamp);
        void onLoadFail(Throwable t);
        void onChangeLike(String feedHash, boolean like);
    }

    void load();
    void loadMore(int page, int timestamp);
    void refresh();
    void changeLike(String feedHash, boolean liked);
    boolean isLiked(String feedHash);
}

File: app/src/main/kotlin/br/com/pedrosilva/tecnonutri/presentation/presenters/FeedPresenter.kt
package br.com.pedrosilva.tecnonutri.presentation.presenters

import com.pedrenrique.tecnonutri.domain.FeedItem
import br.com.pedrosilva.tecnonutri.presentation.presenters.base.BasePresenter
import br.com.pedrosilva.tecnonutri.presentation.ui.BaseView

interface FeedPresenter : BasePresenter {

    interface View : BaseView {
        fun onLoadFeed(
            feedItems: List<FeedItem>,
            page: Int,
            timestamp: Int
        )

        fun onLoadFail(error: Throwable)

        fun onChangeLike(feedHash: String, like: Boolean)
    }

    fun load()

    fun loadMore(page: Int, timestamp: Int)

    fun refresh()

    fun changeLike(feedHash: String, liked: Boolean)

    fun isLiked(feedHash: String): Boolean
}